src/Sulu/Bundle/AdminBundle/Resources/js/containers/Toolbar/Item.js

```diff
@@ -1,28 +1,84 @@
 // @flow
-import Icon from '../../components/Icon';
-import type {Item as ItemType} from './types';
+import {action, observable} from 'mobx';
 import React from 'react';
+import classNames from 'classnames';
+import {observer} from 'mobx-react';
+import type {Item as ItemType} from './types';
+import Icon from '../../components/Icon';
 import itemStyles from './item.scss';
+import Dropdown from './Dropdown';
 
+@observer
 export default class Item extends React.PureComponent {
     props: ItemType;
 
     static defaultProps = {
         enabled: true,
     };
 
+    @observable hasDropdownOpen = false;
+
+    @action toggleDropdown = () => {
+        this.hasDropdownOpen = !this.hasDropdownOpen;
+    };
+
+    @action closeDropdown = () => {
+        if (this.hasSubItems) {
+            this.hasDropdownOpen = false;
+        }
+    };
+
+    componentWillMount() {
+        const {items} = this.props;
+
+        this.hasSubItems = items && items.length;
+    }
+
     handleClick = () => {
-        if (this.props.enabled && this.props.onClick) {
-            this.props.onClick();
+        const {enabled, onClick} = this.props;
+
+        if (enabled) {
```

Comment:
We could do an early return instead of nesting the `if`s:

```javascript
if (enabled) {
    return;
}

if (onClick) {
    onClick();
}

if (this.hasSubItems) {
    this.toggleDropdown();
}